owncloudApp/src/androidTest/java/com/owncloud/android/flow/CreatePublicShareTest.kt

```diff
@@ -0,0 +1,370 @@
+        savePublicShare(newPublicShare2)
+
+        // New share properly created
+        sharesLiveData.postValue(
+            DataResult.success(
+                publicShares.take(2)
+            )
+        )
+
+        // Check whether the dialog to create the public share has been properly closed
+        onView(withText(R.string.share_via_link_create_title)).check(doesNotExist())
+        onView(withText(newPublicShare2.name)).check(matches(isDisplayed()))
+
+        /**
+         * 3rd public share
+         */
+        onView(withId(R.id.addPublicLinkButton)).perform(click())
+
+        val newPublicShare3 = publicShares[2]
+        savePublicShare(newPublicShare3)
+
+        // New share properly created
+        sharesLiveData.postValue(
+            DataResult.success(
+                publicShares
+            )
+        )
+
+        // Check whether the dialog to create the public share has been properly closed
+        onView(withText(R.string.share_via_link_create_title)).check(doesNotExist())
+        onView(withText(newPublicShare3.name)).check(matches(isDisplayed()))
+    }
+
+    @Test
+    fun createShareLoading() {
+        loadCapabilitiesSuccessfully()
+        loadSharesSuccessfully(arrayListOf())
+
+        onView(withId(R.id.addPublicLinkButton)).perform(click())
+
+        savePublicShare(publicShares[0], DataResult.loading())
+
+        onView(withText(R.string.common_loading)).check(matches(isDisplayed()))
+    }
+
+    @Test
+    fun createShareError() {
+        loadCapabilitiesSuccessfully()
+        loadSharesSuccessfully(arrayListOf())
+
+        onView(withId(R.id.addPublicLinkButton)).perform(click())
+
+        savePublicShare(
+            publicShares[0],
+            DataResult.error(
+                RemoteOperationResult.ResultCode.FORBIDDEN,
+                exception = Exception("Error when retrieving shares")
+            )
+        )
+
+        onView(withText(R.string.share_link_file_error)).check(matches(isDisplayed()))
+    }
+
+    private fun getOCFileForTesting(name: String = "default") = OCFile("/Photos").apply {
+        availableOfflineStatus = OCFile.AvailableOfflineStatus.NOT_AVAILABLE_OFFLINE
+        fileName = name
+        fileId = 9456985479
+        remoteId = "1"
+        privateLink = "private link"
+    }
+
+    private fun loadCapabilitiesSuccessfully(
+        capability: OCCapabilityEntity = AppTestUtil.createCapability(
+            versionString = "10.1.1",
+            sharingPublicMultiple = CapabilityBooleanType.TRUE.value
+        )
+    ) {
+        capabilitiesLiveData.postValue(
+            DataResult.success(
+                capability
+            )
+        )
+    }
+
+    private fun loadSharesSuccessfully(shares: ArrayList<OCShareEntity> = publicShares) {
+        sharesLiveData.postValue(DataResult.success(shares))
+    }
+
+    private fun savePublicShare(newShare: OCShareEntity, resource: DataResult<Unit> = DataResult.success()) {
+
+        every {
```

Comment:
in this function, you are mixing mocking with performing a click action.... Please, divide it